source/server/admin/admin.cc

```diff
@@ -161,63 +161,67 @@ AdminImpl::AdminImpl(const std::string& profile_path, Server::Instance& server,
-          {"/help", "print out list of admin commands", MAKE_ADMIN_HANDLER(handlerHelp), false,
-           false},
-          {"/hot_restart_version", "print the hot restart compatibility version",
-           MAKE_ADMIN_HANDLER(server_info_handler_.handlerHotRestartVersion), false, false},
-          {"/logging", "query/change logging levels",
-           MAKE_ADMIN_HANDLER(logs_handler_.handlerLogging), false, true},
-          {"/memory", "print current allocation/heap usage",
-           MAKE_ADMIN_HANDLER(server_info_handler_.handlerMemory), false, false},
-          {"/quitquitquit", "exit the server",
-           MAKE_ADMIN_HANDLER(server_cmd_handler_.handlerQuitQuitQuit), false, true},
-          {"/reset_counters", "reset all counters to zero",
-           MAKE_ADMIN_HANDLER(stats_handler_.handlerResetCounters), false, true},
-          {"/drain_listeners", "drain listeners",
-           MAKE_ADMIN_HANDLER(listeners_handler_.handlerDrainListeners), false, true},
-          {"/server_info", "print server version/status information",
-           MAKE_ADMIN_HANDLER(server_info_handler_.handlerServerInfo), false, false},
-          {"/ready", "print server state, return 200 if LIVE, otherwise return 503",
-           MAKE_ADMIN_HANDLER(server_info_handler_.handlerReady), false, false},
-          {"/stats", "print server stats", MAKE_ADMIN_HANDLER(stats_handler_.handlerStats), false,
-           false},
-          {"/stats/prometheus", "print server stats in prometheus format",
-           MAKE_ADMIN_HANDLER(stats_handler_.handlerPrometheusStats), false, false},
-          {"/stats/recentlookups", "Show recent stat-name lookups",
-           MAKE_ADMIN_HANDLER(stats_handler_.handlerStatsRecentLookups), false, false},
-          {"/stats/recentlookups/clear", "clear list of stat-name lookups and counter",
-           MAKE_ADMIN_HANDLER(stats_handler_.handlerStatsRecentLookupsClear), false, true},
-          {"/stats/recentlookups/disable", "disable recording of reset stat-name lookup names",
-           MAKE_ADMIN_HANDLER(stats_handler_.handlerStatsRecentLookupsDisable), false, true},
-          {"/stats/recentlookups/enable", "enable recording of reset stat-name lookup names",
-           MAKE_ADMIN_HANDLER(stats_handler_.handlerStatsRecentLookupsEnable), false, true},
-          {"/listeners", "print listener info",
-           MAKE_ADMIN_HANDLER(listeners_handler_.handlerListenerInfo), false, false},
-          {"/runtime", "print runtime values", MAKE_ADMIN_HANDLER(runtime_handler_.handlerRuntime),
-           false, false},
-          {"/runtime_modify", "modify runtime values",
-           MAKE_ADMIN_HANDLER(runtime_handler_.handlerRuntimeModify), false, true},
-          {"/reopen_logs", "reopen access logs",
-           MAKE_ADMIN_HANDLER(logs_handler_.handlerReopenLogs), false, true},
+          makeHandler("/", "Admin home page", MAKE_ADMIN_HANDLER(handlerAdminHome), false, false),
```

Comment:
note this block of code is mechanistically transformed from its previous state, as we needed to add Handler wrappers to all the legacy callback-based handlers.

In a future PR, `/stats` and likely also `/cluster_info` will be converted to Chunked handlers.